pkg/foundation/cerrors/fatal.go

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cerrors
+
+import (
+	"fmt"
+)
+
+// fatalError is an error type that will differentiate these from other errors that could be retried.
+type fatalError struct {
+	Err error
+}
+
+// NewFatalError creates a new fatalError.
+func NewFatalError(err error) *fatalError {
```

Comment:
@raulb since the struct is not exported now, we can have this method as `FatalError` 👀

Reply:
@maha-hajja LOL, you're right. I just opened https://github.com/ConduitIO/conduit/pull/1817.